test/jdk/java/foreign/TestStubAllocFailure.java

```diff
@@ -31,36 +31,65 @@
  */
 
 import java.lang.foreign.*;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
 import java.io.IOException;
 import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
 
 import org.testng.annotations.Test;
 
-import static org.testng.Assert.assertTrue;
 import static org.testng.Assert.fail;
 
 public class TestStubAllocFailure extends UpcallTestHelper {
 
     @Test
     public void testUpcallAllocFailure() throws IOException, InterruptedException {
         runInNewProcess(UpcallRunner.class, true, List.of("-XX:ReservedCodeCacheSize=3M"), List.of())
-                .assertSuccess();
+                .shouldNotHaveExitValue(0)
+                .shouldNotHaveFatalError();
```

Comment:
Just curious what non-zero exit value is actually expected here and below?

Reply:
Any non-zero exit value is acceptable. The intent here is to check that the process didn't complete normally.

Reply:
A non-zero non-crashing value. Just wondering what that actually would be?

Reply:
For instance, when we exit due to an uncaught exception, the exit value is 1.